lib/internal/http2/compat.js

```diff
@@ -486,4 +520,41 @@ function onServerStream(stream, headers, flags) {
   server.emit('request', request, response);
 }
 
+class Http2PushRequest {
```

Comment:
Will have to give this some more thought. At this point, at least for the early versions of this, I'm not convinced that we should expose push support via the compat API. Still stewing tho